nexus/src/db/error.rs

```diff
@@ -80,64 +80,65 @@ pub fn diesel_pool_result_optional<T>(
     }
 }
 
-/// Converts a Diesel pool error to an external error when operating on a
-/// particular resource, identified by resource type and lookup type
-/// NOTE: If you already have an `authz::ApiResource`, you should use
-/// [`public_error_from_diesel_pool_authz()`] instead.  Eventually, the only
-/// uses of this function should be in the DataStore functions that actually
-/// look up a record for the first time.
-pub fn public_error_from_diesel_pool_lookup(
-    error: PoolError,
-    resource_type: ResourceType,
-    lookup_type: LookupType,
-) -> PublicError {
-    public_error_from_diesel_pool_helper(error, |error| {
-        public_error_from_diesel_lookup_helper(error, || {
-            PublicError::ObjectNotFound {
-                type_name: resource_type,
-                lookup_type,
-            }
-        })
-    })
-}
-
-/// Converts a Diesel pool error to an external error when operating on a
-/// particular resource, identified by ApiResource
-pub fn public_error_from_diesel_pool_authz<R>(
-    error: PoolError,
-    resource: &R,
-) -> PublicError
-where
-    R: crate::authz::ApiResource,
-{
-    public_error_from_diesel_pool_helper(error, |error| {
-        public_error_from_diesel_lookup_helper(error, || resource.not_found())
-    })
-}
-
-/// Converts a Diesel pool error to an external error,
-/// when requested as part of a creation operation
-pub fn public_error_from_diesel_pool_create(
-    error: PoolError,
-    resource_type: ResourceType,
-    object_name: &str,
-) -> PublicError {
-    public_error_from_diesel_pool_helper(error, |error| {
-        public_error_from_diesel_create(error, resource_type, object_name)
-    })
+/// Allows the caller to handle user-facing errors, and provide additional
+/// context which may be used to populate more informative errors.
+///
+/// Note that all operations may return a "Service Unavailable" error if the
```

Comment:
Okay I think I'm being too pedantic now, but you can also get a 503 if we successfully contact the database and _it_ says it can't run the query (e.g., if it gets EIO or something, I'd expect that).  I think these can all produce a 500 ("internal service error") as well.

I don't know that it's worth being this precise but I figured I'd mention it.

Reply:
Updated with a slightly more generic comment. I don't really want to enumerate all possible errors that could be returned, but just give a "heads up" that server errors are always a possibility.